modules/tinymce/src/themes/silver/test/ts/headless/toolbar/SearchableMenuButtonTest.ts

```diff
@@ -0,0 +1,540 @@
+import { ApproxStructure, Assertions, FocusTools, Keyboard, Keys, Mouse, UiFinder } from '@ephox/agar';
+import { AlloyTriggers, GuiFactory, NativeEvents, TestHelpers } from '@ephox/alloy';
+import { describe, it } from '@ephox/bedrock-client';
+import { Fun, Optional } from '@ephox/katamari';
+import { Attribute, SugarDocument, SugarElement } from '@ephox/sugar';
+
+import { renderMenuButton } from 'tinymce/themes/silver/ui/button/MenuButton';
+
+import { fetchMailMergeData } from '../../module/CommonMailMergeFetch';
+import { structMenuWith, structSearchField, structSearchLeafItem, structSearchParentItem, structSearchResultsWith } from '../../module/CommonMenuTestStructures';
+import * as TestExtras from '../../module/TestExtras';
+
+describe('headless.tinymce.themes.silver.toolbar.SearchableMenuButtonTest', () => {
+  const helpers = TestExtras.bddSetup();
+
+  TestHelpers.GuiSetup.bddAddStyles(SugarDocument.getDocument(), [
+    `.tox-menu .tox-collection__item--active {
+      background-color: black;
+    }`,
+    `.tox-menu {
+      background-color: #3f878bd6;
+      color:white;
+      padding: 2em;
+    }`
+  ]);
+
+  const hook = TestHelpers.GuiSetup.bddSetup(
+    (store, _doc, _body) => GuiFactory.build(
+      renderMenuButton(
+        {
+          text: Optional.some('MailMerge'),
+          icon: Optional.none(),
+          tooltip: Optional.none(),
+          onSetup: Fun.constant(Fun.noop),
+          search: Optional.some({ placeholder: Optional.none() }),
+          fetch: fetchMailMergeData({
+            // If a search pattern is present, collapse into one menu
+            collapseSearchResults: true
+          }, store)
+        },
+        'prefix',
+        helpers.backstage(),
+        Optional.none()
+      )
+    )
+
+  );
+
+  const assertAriaConsistent = async () => {
```

Comment:
If this is async then per conventions it should start with `p`, e.g. `pAssertAriaConsistent`.

Reply:
- [ ] rename function